packages/database/src/models/Invoice/Invoice.ts

```diff
@@ -176,6 +176,24 @@ export class Invoice extends Model {
       return;
     }
 
+    // Confirm price list exists
+    const invoicePriceListId = await this.sequelize.models.InvoicePriceList.getIdForPatientEncounter(encounterId);
+    if (!invoicePriceListId) {
+      return;
+    }
+
+    // Confirm invoice product is not configured to be hidden for this price list
+    const invoicePriceListItem = await this.sequelize.models.InvoicePriceListItem.findOne({
```

Comment:
This works well in the end to have the filtering for automatic items in the model so that it's in one place